finagle-core/src/test/scala/com/twitter/finagle/InetResolverTest.scala

```diff
@@ -76,4 +78,36 @@ class InetResolverTest extends FunSuite {
     assert(statsReceiver.stat("lookup_ms")().size > 0)
   }
 
+  test("updates using custom future pool") {
+    class TestPool(latch: CountDownLatch) extends FuturePool {
+      private val delegate = FuturePool.unboundedPool
+      override def apply[T](f: => T): Future[T] = {
+        latch.countDown()
+        delegate(f)
+      }
+    }
+
+    val latch = new CountDownLatch(1) // DnsResolver will use the pool
+    val resolvePool = new TestPool(latch)
+
+    val dnsResolverWithPool = new DnsResolver(DefaultStatsReceiver, resolvePool)
+
+    def resolveLoopback(host: String): Future[Seq[InetAddress]] = {
+      if (host.equals("127.0.0.1")) dnsResolverWithPool(host)
+      else Future.exception(new UnknownHostException())
+    }
+
+    val pollInterval = Duration(100, TimeUnit.MILLISECONDS)
```

Comment:
Since we're not actually using the magic time features, maybe the `Time.withCurrentTimeFrozen(` isn't necessary. Sorry for the misdirection.

Regarding the latch, the `addr.changes.toFuture` call causes the polling event to get canceled after the `Future` resolves, so it never ticks a second time.  Do something to the tune of
```scala
addr.changes.filter(_ != Addr.Pending).respond(addr => println(s"Address: $addr"))
```
adjust your latch to 2, and the test should complete.